src/mbgl/style/expression/distance.cpp

```diff
@@ -137,25 +190,87 @@ double pointsToPointsDistance(const mapbox::geometry::multi_point<double>& point
     return dist;
 }
 
-std::pair<mbgl::optional<IndexRange>, mbgl::optional<IndexRange>> splitRange(const IndexRange& range, bool isLine) {
-    auto size = getRangeSize(range);
-    if (isLine) {
-        if (size == 2) {
-            return std::make_pair(range, nullopt);
+double pointToPolygonDistance(const mapbox::geometry::point<double>& point,
+                              const mapbox::geometry::polygon<double>& polygon,
+                              mapbox::cheap_ruler::CheapRuler& ruler) {
+    if (pointWithinPolygon(point, polygon, true /*trueOnBoundary*/)) return 0.0;
+    double dist = InfiniteDistance;
+    for (const auto& ring : polygon) {
+        const auto nearestPoint = std::get<0>(ruler.pointOnLine(ring, point));
+        dist = std::min(dist, ruler.distance(point, nearestPoint));
```

Comment:
is `dist` guarantied to be positive?

Reply:
`ruler.distance` calculate the square root of the sum of two squared values